pkg/provider/electricity_maps.go

```diff
@@ -64,34 +64,139 @@ func (e *ElectricityMapsClient) GetCarbonIntensity(ctx context.Context, location
 		return nil, errBadStatus(resp)
 	}
 
-	data := &electricityMapsData{}
-	err = json.NewDecoder(resp.Body).Decode(data)
+	type EMHistoryResponse struct {
+		Zone    string
+		History []electricityMapsData
+	}
+
+	historyResponse := &EMHistoryResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&historyResponse)
 	if err != nil {
 		return nil, err
 	}
 
-	validFrom, err := time.Parse(time.RFC3339Nano, data.DateTime)
+	var carbonIntensityPoints []CarbonIntensity
+	var recentDatapoints = NewMostRecentDataPoints()
+
+	for _, dataPoint := range historyResponse.History {
+
+		// We get the most recent value (which is usually estimated)
+		// and the most recent value which is registered (not estimated)
+		// and they wil end up in the recentDatapoints variable
+		recentDatapoints.update(dataPoint)
+
+	}
+	// We need to check if each value exists, because sometimes there are no estimated datapoints
+	// and sometimes there are only estimated datapoints (depending on the location):
+	if recentDatapoints.estimatedFound {
+		estimatedCarbonIntensity, err := toCarbonIntensity(location, recentDatapoints.estimated)
+		if err == nil {
+			carbonIntensityPoints = append(carbonIntensityPoints, *estimatedCarbonIntensity)
+		}
+	}
+	if recentDatapoints.realFound {
+		realCarbonIntensity, err := toCarbonIntensity(location, recentDatapoints.real)
+		if err == nil {
+			carbonIntensityPoints = append(carbonIntensityPoints, *realCarbonIntensity)
+		}
+	}
+
+	return carbonIntensityPoints, nil
+}
+
+// Helper struct to remove clutter in the calling function
+// while finding the latest (and greatest) data points
+type mostRecentDatapoints struct {
```

Comment:
```suggestion
type electricityMapsDatapoints struct {
```

Same here to add the prefix.